core/src/main/cpp/gdn/projection_mesh/projection_mesh_pool.cpp

```diff
@@ -0,0 +1,23 @@
+#include "projection_mesh_pool.h"
+
+namespace gast {
+
+ProjectionMeshPool::ProjectionMeshPool() = default;
+
+ProjectionMeshPool::~ProjectionMeshPool() = default;
+
+RectangularProjectionMesh *ProjectionMeshPool::get_or_create_rectangular_projection_mesh() {
+    if (!rectangular_projection_mesh) {
+        rectangular_projection_mesh = new RectangularProjectionMesh();
```

Comment:
Here and below, these are Godot's resource type so they need to be created using the same pattern. Use the logic used to create the `ExternalResource` as reference.

Reply:
Done.